pkg/util/fedinformer/genericmanager/multi-cluster-manager.go

```diff
@@ -85,14 +86,21 @@ func (m *multiClusterInformerManagerImpl) getManager(cluster string) (SingleClus
 }
 
 func (m *multiClusterInformerManagerImpl) ForCluster(cluster string, client dynamic.Interface, defaultResync time.Duration) SingleClusterInformerManager {
+	// If informer manager already exist, just return
+	if manager, exist := m.getManager(cluster); exist {
+		return manager
+	}
+
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	// If informer manager already exist, just return
+	// check again in case multiple goroutines have passed the initialization check.
 	if manager, exist := m.managers[cluster]; exist {
 		return manager
 	}
 
+	klog.Infof("building informer manager for cluster: %s", cluster)
```

Comment:
such logs seem not required, or increase the log level, otherwise there will be a lot of useless log printing.